keras_preprocessing/image/directory_iterator.py

```diff
@@ -90,7 +90,7 @@ def __init__(self,
                                                             save_format,
                                                             subset,
                                                             interpolation)
-        self.directory = directory
+        self.directory = directory or ''
```

Comment:
In `DirectoryIterator` a the `directory` argument is not optional, no need for the `or` operator.

Reply:
Yes, you are right.
I fixed it